js/src/features/human-resouce/epics.ts

```diff
@@ -392,3 +393,18 @@ export const refetchEmployeeLocationsEpic: RootEpic = (
       ),
     ),
   );
+
+export const fetchVacationsEpic: RootEpic = (action$, state$, { api }) =>
+  action$.pipe(
+    filter(isActionOf(fetchVacationsAsync.request)),
+    switchMap(() =>
+      from(api.humanResource.fetchVacations()).pipe(
+        map((result) =>
+          result.success
+            ? fetchVacationsAsync.success(result.data)
+            : fetchVacationsAsync.failure(result.reason),
+        ),
+        catchError((message) => of(fetchVacationsAsync.failure(message))),
```

Comment:
I think there is no need to add this catch because there is only API request and api errors are already handled in service.

Reply:
`catchError` is also needed for cases when an unexpected error occurred so that there would be no unsubscription from the stream.

Reply:
Removed `catchError`